File: solution/791. Custom Sort String/791_02.cpp
class Solution {
public:
    string customSortString(string order, string str) {
        unordered_map<char, int> m;
        for (char& c : str)
            m[c]++;
        
        string ans;
        for (char& c : order) {
            ans += string(m[c], c);
            m[c] = 0;
        }
        
        for (auto& [c, count] : m)
            ans += string(count, c);
        
        return ans;
    }
};

File: solution/791. Custom Sort String/791_02.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        m = defaultdict(int)
        for c in s:
            m[c] += 1
            
        res = ''
        for c in order:
            res += c * m[c]
            m[c] = 0
        
        for c, count in m.items():
            res += c * count
        
        return res
    
